### DISCUSSION TITLE
How to view ktx files generated by cmgen ?

### DISCUSSION THREAD

**Discussion by Honya2000** [google/filament#6959] [General]
Hello,
I tried all existing viewers to view .ktx ibl files without luck.
Most of them don't support ktx. Found couple of viewers with ktx support, but they don't support cube-maps or R11F_G11F_B10F format.
So currently i'm stuck. How to open those files?

For some reason image-viewer sample doesn't support ktx.

**Comment by bejado** [comment]
Our gltf_viewer sample can load KTX cubemaps.

```
cmgen --format ktx -x cmgenerated/ third_party/environments/venetian_crossroads_2k.hdr
gltf_viewer --ibl cmgenerated
```

**Reply by romainguy** [reply]
There won't be viewers able to handle the format given how we encode R11F_G11F_B10F into RGBA8 data. The best solution is what @bejado proposed.

**Reply by Honya2000** [reply]
cmgen is not able to read those ktx as well.

c:\Projects\bin>cmgen --format ktx -x cmgenerated/ env2.ktx
Decoding image...
Unable to open image: env2.ktx

**Reply by Honya2000** [reply]
Ok, you mean i can see my ktx background in gltf_viewer. But this is not an image viewer. 
I expexted to directly view ktx cubemap file or through some intermediate conversion with cmgen.
But both ways aren't working.
Will modify image-viewer to show ktx cubemaps in cross-layout...
